scripts/cybershake/cybershake_progress.py

```diff
@@ -0,0 +1,364 @@
+        num_all_rels = int(
+            progress_df.loc[fault, ("total", R_COUNT_COL)]
+        )  # not sure why this is np.float64
+        num_complete_rels = []
+        usage_str_list = []
+        is_complete = {}
+        for proc_type in PROCESS_TYPES:
+            usage_str_list.append(get_usage_str(fault, proc_type))
+            num_complete = get_completed_r_count(db_file, proc_type, fault)
+            num_complete_rels.append(f"{num_complete}/{num_all_rels}")
+            is_complete[proc_type] = 1 if num_complete == num_all_rels else 0
+
+        output_list = np.array(
+            list(zip(usage_str_list, num_complete_rels, is_complete.values()))
+        ).flatten()
+        f.write(template_fmt.format(fault, *output_list, get_usage_str(fault, "total")))
+        f.write("\n")
+
+        for proc_type in PROCESS_TYPES:
+            if is_complete[proc_type]:
+                est_completed_all[fault][proc_type] = progress_df.loc[
+                    fault, (proc_type, EST_CORE_HOURS_COL)
+                ]
+
+    est_completed_df = pd.DataFrame.from_dict(
+        est_completed_all, orient="index", columns=PROCESS_TYPES
+    )
+
+    overall_df = progress_df.sum()
+
+    completed_r_counts = {}
+    for proc_type in PROCESS_TYPES:
+        completed_r_counts[proc_type] = get_completed_r_count(db_file, proc_type)
+
+    #    total_r_completed = progress_df["total", COMPLETED_R_COUNT_COL].sum()
+    total_r_count = progress_df["total", R_COUNT_COL].sum()
+
+    f.write("\nOverall progress\n")
+    template_fmt = "{:<6} : {:>5}/{:>5} ({:4.2f}%) RELs complete: {:10.2f} CH consumed / {:10.2f} CH est ({: >#06.2f}%) {:10.2f} CH remaining est\n"
+    for proc_type in PROCESS_TYPES:
+        f.write(
+            template_fmt.format(
+                proc_type,
+                completed_r_counts[proc_type],
+                total_r_count,
+                completed_r_counts[proc_type] / total_r_count * 100,
+                overall_df[proc_type, ACT_CORE_HOURS_COL],
+                overall_df[proc_type, EST_CORE_HOURS_COL],
+                overall_df[proc_type, ACT_CORE_HOURS_COL]
+                / overall_df[proc_type, EST_CORE_HOURS_COL]
+                * 100,
+                overall_df[proc_type, EST_CORE_HOURS_COL]
+                - est_completed_df[proc_type].sum(),
+            )
+        )
+
+    f.close()
+
+
+def send2slack(msg_file, users, url):
+    with open(msg_file, "r") as f:
+        lines = f.readlines()
+
+    summary = (
+        " ".join(users) + " " + "\n".join(lines[-4:])
+    )  # .encode('utf-8') #last 4 lines
+    print(summary)
+    data = json.dumps({"text": summary})
+    urlopen(url, data=data.encode("utf-8"))
+
+
+def main(root_dir: Path, cybershake_list, outfile, slack_config=None):
+
+    runs_dir = Path(sim_struct.get_runs_dir(root_dir))
+
+    fault_names, r_counts = get_faults_and_r_count(cybershake_list)
+
+    # Sort, as faults are run in alphabetical order
```

Comment:
Not always, as you can see by the sample output above.

Reply:
True - This comment is historical. Removing it.